playbooks/init/basic_facts.yml

```diff
@@ -28,27 +28,22 @@
 
 - name: Read API URL from infra config
   hosts: "{{ l_init_fact_hosts | default('nodes') }}"
-  roles:
-  - lib_openshift
   tasks:
-  - oc_obj:
-      name: cluster
-      kind: infrastructures.config.openshift.io
+  - name: Read cluster config
+    k8s_facts:
       kubeconfig: "{{ kubeconfig_path }}"
-      state: present
+      kind: Infrastructure
+      name: cluster
     register: clustercfg
     when: kubeconfig_path is defined
     until:
-    - clustercfg.results is defined
-    - clustercfg.results.returncode is defined
-    - clustercfg.results.returncode == 0
-    - clustercfg.results.results is defined
-    - clustercfg.results.results | length > 0
+    - clustercfg.resources is defined
+    - clustercfg.resources | length > 0
```

Comment:
Check that `status.apiServerURL` exists - operator may not have filled this yet

Reply:
Pushed a new commit to address this.